lib/puppet/type/user.rb

```diff
@@ -227,6 +227,9 @@ def change_to_s(currentvalue, newvalue)
         * OS X 10.8 and higher use salted SHA512 PBKDF2 hashes. When managing passwords
           on these systems, the `salt` and `iterations` attributes need to be specified as
           well as the password.
+        * OS X 10.15 and higher require the salt to be 32-bytes. Since Puppet's user 
```

Comment:
This should be "macOS" instead of "OS X" as well

Reply:
Shoot, I completely forgot that. Thank you!